Source/WebCore/html/HTMLMaybeFormAssociatedCustomElement.h

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright (C) 2022 Apple Inc. All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ * 1. Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in the
+ *    documentation and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY APPLE INC. AND ITS CONTRIBUTORS ``AS IS''
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO,
+ * THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+ * PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL APPLE INC. OR ITS CONTRIBUTORS
+ * BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF
+ * THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#pragma once
+
+#include "FormAssociatedCustomElement.h"
+#include "LabelableElement.h"
+#include <JavaScriptCore/JSGlobalObject.h>
+
+namespace WebCore {
+
+class HTMLMaybeFormAssociatedCustomElement final : public LabelableElement {
```

Comment:
We should probably rename LabelableElement to HTMLLabelableElement in a separate patch.

Reply:
`(HTML)LabelableElement` name is kinda misleading IMO: it's actually labelable only if `supportLabels` is true, which we shouldn't forget to check after `is<LabelableElement>`.

@rniwa would you consider a patch that removes `LabelableElement` altogether and moves

```
    WEBCORE_EXPORT RefPtr<NodeList> labels();
    virtual bool supportLabels() const { return false; }
```

up to `HTMLElement`? That would also likely reduce virtual function calls in `HTMLLabelElement`.

Reply:
Yeah, I don't think there is much value in having this class in C++. We should probably do that in a separate patch though.

Reply:
PFR: https://github.com/WebKit/WebKit/pull/8619